src/Drawable.js

```diff
@@ -373,10 +373,15 @@ class Drawable {
     /**
      * Check if the world position touches the skin.
      * @param {twgl.v3} vec World coordinate vector.
+     * @param {boolean} ignoreInvisibility If true, this function will not early exit if
+     * this drawable is invisible. Defaults to false.
      * @return {boolean} True if the world position touches the skin.
      */
-    isTouching (vec) {
-        if (!(this.skin && this._visible)) {
+    isTouching (vec, ignoreInvisibility) {
+        if (!this.skin) {
+            return false;
+        }
+        if (!(ignoreInvisibility || this._visible)) {
```

Comment:
I think we might be able to just remove the _visible check from here, it's kind of redundant and happens inside of a few tight loops anyway -- we can do this anywhere Drawable.isTouching gets called - ensure it's visible first..